docs/5.0/preview/database-access.md

```diff
@@ -191,30 +286,140 @@ proxy_service:
+section above. See [Creating and using an IAM policy for IAM database access](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html)
+in AWS documentation.
+
+Teleport database service uses default credential provider chain to find AWS
+credentials. See [Specifying Credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials)
+for more information.
+
+## Connect
 
-After the setup, log into the cluster and see the registered database proxy:
+Once the database service has joined the cluster, login to see the available
+databases:
 
 ```sh
 $ tsh login --proxy=teleport.example.com:3080
 $ tsh db ls
+Name   Description Labels
+------ ----------- --------
+aurora AWS Aurora  env=dev
 ```
 
-### Connect
+Note that you will only be able to see databases your role has access to. See
+[RBAC](#rbac) section for more details.
 
-Log into the database and connect using `psql`:
+To connect to a particular database server, first retrieve credentials from
+Teleport using `tsh db login` command:
 
 ```sh
-$ tsh db login postgres
-$ psql "service=postgres-aurora user=<db-user> database=<db-name>"
+$ tsh db login aurora
 ```
 
+!!! tip
+    You can be logged into multiple databases simultaneously.
+
+You can optionally also specify the database name and the user to use by
+default when connecting to the database instance:
+
+```sh
+$ tsh db login --db-user=postgres --db-name=postgres aurora
+```
+
+When logging into a PostgreSQL database, `tsh` automatically configures a section
+in the [connection service file](https://www.postgresql.org/docs/current/libpq-pgservice.html)
+with the name of `<cluster-name>-<database-service-name>`.
+
+Suppose the cluster name is "root", then you can connect to the database using
+the following `psql` command:
+
+```sh
+# Use default database user/name.
+$ psql "service=root-aurora"
+# Specify database name/user explicitly.
+$ psql "service=root-aurora user=alice dbname=metrics"
+```
+
+To log out of the database and remove credentials:
+
+```sh
+# Log out of a particular database instance.
+$ tsh db logout aurora
+# Log out of all database instances.
+$ tsh db logout
+```
+
+## RBAC
+
+Teleport's "role" resource provides the following instruments for restricting
+database access:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: developer
+spec:
+  allow:
+    # Label selectors for database instances this role has access to. These
+    # will be matched against the static/dynamic labels set on the database
+    # service.
+    db_labels:
+      environment: ["dev", "stage"]
+    # Database names this role will be able to connect to. Note: this is not
+    # the same as the "name" field in "db_service", this is the database names
+    # within a particular database instance.
+    db_names: ["main", "metrics", "postgres"]
+    # Database users this role can connect as.
+    db_users: ["alice", "bob"]
```

Comment:
It would be good to add add a deny example, and possibly highlight more advanced features such as wildcards. 

e.g. 

```deny:
      db_name: ["prod*"]

Reply:
Added below.